tests/utils/test_qtutils.py

```diff
@@ -91,3 +125,836 @@ def test_qt_both(self, parser):
+
+    """Tests for serialize_stream and deserialize_stream."""
+
+    def _set_status(self, stream, status):
+        """Helper function so mocks can set an error status when used."""
+        stream.status.return_value = status
+
+    @pytest.fixture
+    def stream_mock(self):
+        """Fixture providing a QDataStream-like mock."""
+        m = unittest.mock.MagicMock(spec=QDataStream)
+        m.status.return_value = QDataStream.Ok
+        return m
+
+    def test_serialize_pre_error_mock(self, stream_mock):
+        """Test serialize_stream with an error already set."""
+        stream_mock.status.return_value = QDataStream.ReadCorruptData
+
+        with pytest.raises(OSError) as excinfo:
+            qtutils.serialize_stream(stream_mock, QPoint())
+
+        assert not stream_mock.__lshift__.called
+        assert str(excinfo.value) == "The data stream has read corrupt data."
+
+    def test_serialize_post_error_mock(self, stream_mock):
+        """Test serialize_stream with an error while serializing."""
+        obj = QPoint()
+        stream_mock.__lshift__.side_effect = lambda _other: self._set_status(
+            stream_mock, QDataStream.ReadCorruptData)
+
+        with pytest.raises(OSError) as excinfo:
+            qtutils.serialize_stream(stream_mock, obj)
+
+        assert stream_mock.__lshift__.called_once_with(obj)
+        assert str(excinfo.value) == "The data stream has read corrupt data."
+
+    def test_deserialize_pre_error_mock(self, stream_mock):
+        """Test deserialize_stream with an error already set."""
+        stream_mock.status.return_value = QDataStream.ReadCorruptData
+
+        with pytest.raises(OSError) as excinfo:
+            qtutils.deserialize_stream(stream_mock, QPoint())
+
+        assert not stream_mock.__rshift__.called
+        assert str(excinfo.value) == "The data stream has read corrupt data."
+
+    def test_deserialize_post_error_mock(self, stream_mock):
+        """Test deserialize_stream with an error while deserializing."""
+        obj = QPoint()
+        stream_mock.__rshift__.side_effect = lambda _other: self._set_status(
+            stream_mock, QDataStream.ReadCorruptData)
+
+        with pytest.raises(OSError) as excinfo:
+            qtutils.deserialize_stream(stream_mock, obj)
+
+        assert stream_mock.__rshift__.called_once_with(obj)
+        assert str(excinfo.value) == "The data stream has read corrupt data."
+
+    def test_round_trip_real_stream(self):
+        """Test a round trip with a real QDataStream."""
+        src_obj = QPoint(23, 42)
+        dest_obj = QPoint()
+        data = QByteArray()
+
+        write_stream = QDataStream(data, QIODevice.WriteOnly)
+        qtutils.serialize_stream(write_stream, src_obj)
+
+        read_stream = QDataStream(data, QIODevice.ReadOnly)
+        qtutils.deserialize_stream(read_stream, dest_obj)
+
+        assert src_obj == dest_obj
+
+    @pytest.mark.qt_log_ignore('^QIODevice::write: ReadOnly device')
```

Comment:
I suggest to add `pytest-qt>=1.4` to your `tox.ini` (at least I don't see this change in this PR).